apps/rxjs-race-condition/src/app/app.component.ts

```diff
@@ -19,12 +19,12 @@ export class AppComponent implements OnInit {
   title = 'rxjs-race-condition';
   dialog = inject(MatDialog);
   topicService = inject(TopicService);
-  topics: TopicType[] = [];
+  topics: TopicType[] | string[] = [];
 
   ngOnInit(): void {
     this.topicService
       .fakeGetHttpTopic()
-      .pipe(take(1))
+      .pipe(take(1), startWith(['Politic', 'Culture', 'Nature']))
```

Comment:
That's not the solution. Indeed you will get data inside your topics array, but not the right one. 

The goal is to see that imperative coding can create race condition. Refactor the code to declarative coding.

Reply:
Thanks, I'll have another go

Reply:
@tomalaforge Hey I've done a new commit which has a declarative solution now.